KAS-2407: column selector modal

MikiDi opened the pull request:
See [KAS-2407](https://kanselarij.atlassian.net/browse/KAS-2407). Refactors the publications overview table column-toggle modal. Different columns are now sourced from a config file. Slightly out of scope of this story, also the initialization of the "show-column-flags" as well as the rendering of the column headers is now sourced from this same config-file.  
Sourcing the headers from the config file also involved some changes to the sorting logic. The sorting now works as before, but please note that the sorting for some columns is still buggy. The error thrown when sorting by "uiterste publicatie" for example is out of scope of this story.

MikiDi committed:
refactor: move table-display-config-modal into separate component

MikiDi committed:
refactor: move columns to config file + render dynamically

MikiDi committed:
refactor: add column default display-settings to config-file

MikiDi committed:
refactor: DRYer column names

MikiDi committed:
refactor: rename variable for clarity

MikiDi committed:
refactor: rename variable for clarity

MikiDi committed:
refactor: render table headers dynamically based on column config

MikiDi committed:
chore: remove dangling code comment

ValenberghsSven reviewed (changes requested):
@MikiDi 
Deze zal nog moeten aangepast worden naar aanleiding van een recente change: "derived publication type" => "regulation type"

MikiDi committed:
Merge branch 'development' into feature/KAS-2407-column-selector-modal

MikiDi commented:
Done @ValenberghsSven

erikap committed:
Merge branch 'development' into feature/KAS-2407-column-selector-modal

erikap committed:
Remove local copy of options in overview-table-display-config-modal

Since objects are passed by reference using this.options doesn't have
an added value. Since making changes to the checkboxes in the modal
has immediate effect (no save button to confirm) it's OK to directly
changes the properties of the passed options object

erikap committed:
Merge branch 'development' into feature/KAS-2407-column-selector-modal

erikap committed:
Merge branch 'development' into feature/KAS-2407-column-selector-modal

erikap committed:
Merge branch 'development' into feature/KAS-2407-column-selector-modal

erikap committed:
Fix linting
